pkg/querier/worker/scheduler_processor.go

```diff
@@ -158,6 +158,10 @@ func (sp *schedulerProcessor) querierLoop(c schedulerpb.SchedulerForQuerier_Quer
 			logger := util_log.WithContext(ctx, sp.log)
 			sp.runRequest(ctx, logger, request.QueryID, request.FrontendAddress, request.StatsEnabled, request.HttpRequest)
 
+			if err = ctx.Err(); err != nil {
```

Comment:
This looks incorrect to me. If we return here, we'll not communicate back to the scheduler in line 165?

Reply:
When the upstream request is cancelled, the scheduler would have handled it first already. So I thought there is no point of querier communicating back to the scheduler, as it would just throw error anyways. Here's how scheduler handles the upstream request cancellation: https://github.com/cortexproject/cortex/blob/4e162a011a5e23f90c0c934c3d0dee6398d349fd/pkg/scheduler/scheduler.go#L430-L434